oaknational/Oak-Web-Application
# Feat/lesq 1006/onboarding role selection [LESQ-1006]

## Description

Music year: {owa_music_year}

- refactors onboarding page into `role-selection` and `school-selection` sub routes
- extracts onboarding form into wrapper component
- create schema file for form types
- create new form for role selection

## How to test

With the feature flag enabled

1. Go to https://deploy-preview-2691--oak-web-application.netlify.thenational.academy/onboarding/role-selection and /school-selection
2. Click on the options, including other, see error states etc
3. There is a console log on successful submit
4. The newsletter checkbox should still be present, as well as Ts and Cs, across both onboarding pages
5. The journey should work from `/onboarding` -> select no -> arrive at `/onboarding/role-selection`

## Screenshots
<img width="500" alt="Screenshot 2024-08-13 at 11 18 00" src="https://github.com/user-attachments/assets/e23009a6-9600-4f29-b902-3df8d50fbde5">


## Checklist

- [ ] Nothing selected by default, if a user selects “Continue” without selecting an option, error should say 'select a role'
- [ ] If other is selected, required text input displayed
- [ ] If a use selects continue without inputting a role into the text box, error should say 'please tell us what your role is'
- [ ] error should clear when a role is selected or input
- [ ] newsletter checkbox toggles
